ibm/mas_devops/roles/mongodb/tasks/providers/community.yml

```diff
@@ -60,10 +60,69 @@
     definition: "{{ lookup('template', 'templates/community/manager.yml') }}"
 
 
-# 5. Configure TLS
+# 5. Configure TLS using cert manager
 # -----------------------------------------------------------------------------
-# TODO: We should be able to do this with cert-manager natively inside Kubernetes
+#This task will create ClusterIssuer and wait till it gets ready
```

Comment:
Where/how are all these new certs being used? 

- We create a cluster issuer `selfsigned-clusterissuer`
- We then use that cluster issuer to issue a certificate `mongo-ca-crt`
- We then create an issuer `mongo-issuer` signed by the `mongo-ca-crt`
- We now use the `mongo-issuer` to create a server cert `mongo-server` and a client cert `mongo-client`

Why do we need a clusterissuer if we're only using it in the context of this creating a mongo CA cert/this namespace?

None of the code that set up mongo has changed ... how do these new certs actually get used, all the original code that sets up certificates using openssl shell commands is still in the task as well, and the mongo install seems to still be using the secrets created by those steps.

Maybe I'm misreading the code, but I don't see where these certs fit into the picture:

```
  security:
    tls:
      enabled: true
      certificateKeySecretRef:
        name: mas-mongo-ce-cert-secret
      caConfigMapRef:
        name: mas-mongo-ce-cert-map
```

This still tells Mongo to use the original certs that are created by the pre-existing code (in `templates/community/tls.yml`), what's using these new certs & how?

Reply:
1.Clusterissuer is being used by mongo CA cert as issuer reference and same Clusterissuer can be used by other applications.
2.We have added tasks to use cert manager in generated certificates to be used by mongodb cluster

Reply:
Hi @whitfiea , We have made changes suggested by you regarding Issuer and CA cert.
Also we are referring to the correct server certificate secret in cr.yml.
Thanks for your inputs.